app/src/services/cpk.ts

```diff
@@ -274,6 +274,165 @@ class CPKService {
+
+      const openingDateMoment = moment(resolution)
+
+      const scalarLow = lowerBound && Number(formatBigNumber(lowerBound, 18))
+      const scalarHigh = upperBound && Number(formatBigNumber(upperBound, 18))
+      const scalarStart = startingPoint && Number(formatBigNumber(startingPoint, 18))
+
+      const transactions = []
+
+      let questionId: string
+      if (loadedQuestionId) {
+        questionId = loadedQuestionId
+      } else {
+        // Step 1: Create question in realitio without bounds
+        transactions.push({
+          to: realitioAddress,
+          data: RealitioService.encodeAskScalarQuestion(
+            question,
+            unit,
+            category,
+            arbitrator.address,
+            openingDateMoment,
+            networkId,
+          ),
+        })
+        questionId = await realitio.askScalarQuestionConstant(
+          question,
+          unit,
+          category,
+          arbitrator.address,
+          openingDateMoment,
+          networkId,
+          this.cpk.address,
+        )
+      }
+      logger.log(`QuestionID ${questionId}`)
+
+      // Step 1.5: Announce the questionId and its bounds to the RealitioScalarAdapter
+      scalarLow &&
+        scalarHigh &&
+        transactions.push({
+          to: realitioScalarAdapterAddress,
+          data: RealitioService.encodeAnnounceConditionQuestionId(questionId, scalarLow, scalarHigh),
+        })
+
+      const oracleAddress = getContractAddress(networkId, 'realitioScalarAdapter')
+      const conditionId = conditionalTokens.getConditionId(questionId, oracleAddress, 2)
+
+      let conditionExists = false
+      if (loadedQuestionId) {
+        conditionExists = await conditionalTokens.doesConditionExist(conditionId)
+      }
+
+      if (!conditionExists) {
+        // Step 2: Prepare scalar condition using the conditionQuestionId
+        logger.log(`Adding prepareCondition transaction`)
+
+        scalarLow &&
+          scalarHigh &&
+          transactions.push({
+            to: conditionalTokensAddress,
+            data: ConditionalTokenService.encodePrepareScalarCondition(
+              questionId,
+              scalarLow,
+              scalarHigh,
+              oracleAddress,
+            ),
+          })
+      }
+
+      logger.log(`ConditionID: ${conditionId}`)
+
+      // Step 3: Approve collateral for factory
+      transactions.push({
+        to: collateral.address,
+        data: ERC20Service.encodeApproveUnlimited(marketMakerFactory.address),
+      })
+
+      // Step 4: Transfer funding from user
+      transactions.push({
+        to: collateral.address,
+        data: ERC20Service.encodeTransferFrom(account, this.cpk.address, marketData.funding),
+      })
+
+      // Step 4.5: Calculate probabilities for distributionHint
+      const a = scalarLow && scalarHigh && scalarStart && ((scalarStart - scalarLow) / (scalarHigh - scalarLow)) * 100
```

Comment:
I would write it such that if `scalarStart` exists, then

```js
// this might be a useful utility to have somewhere
function clampBigNumber(x, min, max) {
  if (x.lt(min)) return min;
  if (x.gt(max)) return max;
  return x;
}

const domainSize = scalarHigh.sub(scalarLow)

a = clampBigNumber(scalarHigh.sub(scalarStart), ethers.constants.Zero, domainSize)
b = clampBigNumber(scalarStart.sub(scalarLow), ethers.constants.Zero, domainSize)

// scaling to 100 is optional if distributionHint is calculated like below
// but these lines would do it, just for reference:
// a = Number(a.toString()) * 100 / Number(domainSize.toString());
// b = Number(b.toString()) * 100 / Number(domainSize.toString());
```

In fact, we can get the `distributionHint` directly, skipping `calcDistributionHint`:

```
const distributionHint = [b, a];
```